src/POData/BaseService.php

```diff
@@ -433,10 +433,12 @@ protected function serializeResult(RequestDescription $request, IUriProcessor $u
                 if ($request->isLinkUri()) {
                     // In the query 'Orders(1245)/$links/Customer', the targeted
                     // Customer might be null
-                    if (null === $result->results) {
+                    if (null === $result->results && HTTPRequestMethod::POST() != $method && HTTPRequestMethod::DELETE() != $method) {
```

Comment:
Can you replace HTTPRequestMethod::POST() != $method    by $methodIsNotPost?  Maybe do something similar for the delete verb check?